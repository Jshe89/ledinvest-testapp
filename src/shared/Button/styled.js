import styled from 'styled-components';
import palette from '../../styles/palette';

export const Button = styled.button`
  border: 1px solid ${palette.bgColors.yellowDark};
  background: ${palette.bgColors.yellow};
  text-transform: uppercase;
  color: ${palette.textColors.defaultColor};
  padding: 10px 25px;
  cursor: pointer;
  outline: none;
  &:hover {
    background: ${palette.bgColors.yellowDark};
    font-weight: bold;
  }
  &:disabled {
    opacity: 0.5;
    pointer-events: none;
  }
`;
