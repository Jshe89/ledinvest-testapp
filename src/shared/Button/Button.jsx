import React from 'react';
import { Button as ButtonStyled } from './styled';

const Button = props => <ButtonStyled {...props}>{props.children}</ButtonStyled>;

export default Button;
