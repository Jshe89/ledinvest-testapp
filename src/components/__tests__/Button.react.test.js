import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../shared/Button';

test('Button rendered', () => {
  const btn = renderer.create(<Button>invest</Button>);
  let tree = btn.toJSON();
  expect(tree).toMatchSnapshot();
});
