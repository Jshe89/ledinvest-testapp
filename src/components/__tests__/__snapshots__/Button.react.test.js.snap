// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered 1`] = `
<button
  className="sc-bdfBwQ fwycA-D"
>
  invest
</button>
`;
